tests/rptest/tests/audit_log_test.py

```diff
@@ -1911,9 +1953,52 @@ def __init__(self, test_context):
                                      }),
             schema_registry_config=sr_config)
 
+    def match_authn_record(self, record, status_id: StatusID):
+        if record['class_uid'] == ClassUID.AUTHENTICATION and record[
+                'dst_endpoint']['svc_name'] == self.sr_audit_svc_name:
+            self.logger.debug(f"Validating auth record: {record}")
+
+        return record['class_uid'] == ClassUID.AUTHENTICATION and \
+            record['dst_endpoint']['svc_name'] == self.sr_audit_svc_name and \
+            record['user']['name'] == self.username and \
+            record['status_id'] == status_id
+
+    def match_api_record(self,
+                         record,
+                         endpoint,
+                         status_id: Optional[StatusID] = None):
+        if record['class_uid'] == ClassUID.API_ACTIVITY and \
+            record['dst_endpoint']['svc_name'] == self.sr_audit_svc_name:
+            self.logger.debug(f"Validating api activity record: {record}")
+
+        if status_id and record.get('status_id', '') != status_id:
+            return False
+
+        if record['class_uid'] == ClassUID.API_ACTIVITY \
+            and record['dst_endpoint']['svc_name'] == self.sr_audit_svc_name \
+            and record['actor']['user']['name'] == self.username:
+            regex = re.compile(
+                "http:\/\/(?P<address>.*):(?P<port>\d+)\/(?P<handler>.*)")
+            url_string = record['http_request']['url']['url_string']
+            match = regex.match(url_string)
+            if match and match.group('handler') == endpoint:
+                return True
+
+        return False
```

Comment:
nitpick: I think it's clearer with early exit (I also took the liberty of using a raw string for the regex):
```suggestion
        if record['class_uid'] != ClassUID.API_ACTIVITY or record[
                'dst_endpoint']['svc_name'] != self.sr_audit_svc_name:
            return False

        self.logger.debug(f"Validating api activity record: {record}")
        if (status_id and record.get('status_id', '') != status_id
            ) or record['actor']['user']['name'] != self.username:
            return False

        regex = re.compile(
            r"http://(?P<address>.*):(?P<port>\d+)/(?P<handler>.*)")
        url_string = record['http_request']['url']['url_string']
        match = regex.match(url_string)
        return match and match.group('handler') == endpoint
```